core/trino-spi/src/main/java/io/trino/spi/ptf/TableArgument.java

```diff
@@ -102,6 +102,71 @@ public boolean isPassThroughColumns()
         return passThroughColumns;
     }
 
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    public static final class Builder
+    {
+        private Optional<QualifiedName> name;
+        private RowType rowType;
+        private List<String> partitionBy = List.of();
+        private List<SortItem> orderBy = List.of();
+        private boolean rowSemantics;
+        private boolean pruneWhenEmpty;
+        private boolean passThroughColumns;
+
+        private Builder() {}
+
+        public Builder name(Optional<QualifiedName> name)
+        {
+            this.name = name;
+            return this;
+        }
+
+        public Builder rowType(RowType rowType)
+        {
+            this.rowType = rowType;
+            return this;
+        }
+
+        public Builder partitionBy(List<String> partitionBy)
+        {
+            this.partitionBy = partitionBy;
+            return this;
+        }
+
+        public Builder orderBy(List<SortItem> orderBy)
+        {
+            this.orderBy = orderBy;
+            return this;
+        }
+
+        public Builder rowSemantics(boolean rowSemantics)
+        {
+            this.rowSemantics = rowSemantics;
+            return this;
+        }
+
+        public Builder pruneWhenEmpty(boolean pruneWhenEmpty)
```

Comment:
Generally I prefer boolean methods to have a good default and then a no-arg method to change tha behavior, so I would make this:
```
public Builder pruneWhenEmpty()
{
    this.pruneWhenEmpty = true;
    return this;
}

Reply:
It was the initial approach. I changed it according to https://github.com/trinodb/trino/pull/12350#discussion_r872280705